jmclient/jmclient/blockchaininterface.py

```diff
@@ -185,26 +348,23 @@ def __init__(self, jsonRpc, network, wallet_name):
                     "setting in joinmarket.cfg) instead. See docs/USAGE.md "
                     "for details.")
 
-    def is_address_imported(self, addr):
+    def is_address_imported(self, addr: str) -> bool:
         return len(self._rpc('getaddressinfo', [addr])['labels']) > 0
 
-    def get_block(self, blockheight):
+    def get_block(self, blockheight: int) -> Optional[str]:
```

Comment:
Don't see any reason for Optional here. The function misses error checking (what if get block hash fails), but that's orthogonal to typehints, also, this has only ever been used for blockchain scanning so far.

I suppose there's an argument for Optional based on an idea of None being used as the return type for failure? But that's just imagined, not actual. I'd just have str only, since that's the function's intent, and exception classes can handle failures.

Reply:
Agree, addressed in 094df20c7840ce1cbb27dff08f48c0caa69a17dc.